src/js/bindings/publisher.js

```diff
@@ -0,0 +1,23 @@
+// function takes a callback and executes when checkboxes under #hero-filter-publisher is changed.
+export function bindCharacterFilterPublisher(callback) {
+  callback();
+  const allCheckboxes = document.querySelectorAll('#hero-filter-publisher input[type="checkbox"]');
+  // loop over checkboxes w/ event listener to detect if they are checked
+  allCheckboxes.forEach((checkbox) => checkbox.addEventListener('change', () => {
+    callback();
+  }));
+}
+
+// filters from checkboxes
+export function getSelectedFilterPublisher() {
+// selects whatever has hero-filter-publisher id and has checkbox input
+  const allCheckboxes = document.querySelectorAll('#hero-filter-publisher input[type="checkbox"]');
+  // eslint-disable-next-line max-len
+  // from copy of array of items returned above, filters for checked item, returns name of publisher (element.defaultValue)
+  const selectedFilterPublisher = [...allCheckboxes]
```

Comment:
```
  const selectedFilterPublisher = [...allCheckboxes]
    .filter((element) => element.checked === true)
    .map((element) => element.defaultValue);

```
I am a big fan of this pattern and I think its great you included it here. Here is a fun little bit a knowledge for future use. It appears you are using the spread operator to create a copy of allCheckboxes `[...allCheckboxes]`. That is a great thing to do because other wise we might end up with some unintended side affects. But in this particular situation you do not need to explicitly create a copy of the array. The reason for this is that the `.filter()` method returns a copy of the array it was called on. So if you look at the two examples below they both do the same thing. 

```
const b = [...a].filter(some filter function);
// is the same as
const b = a.filter(some filter function)
```